openlcsd/flow/tasks.py

```diff
@@ -1605,23 +1606,19 @@ def trigger_corgi_components_imports(context, engine):
     engine.logger.info("Done.")
 
 
-# sub-flow of `flow_get_corgi_components`
-process_collected_components = [
+flow_collect_components_for_subscription = [
+    get_config,
+    populate_components_generator,
     populate_source_components,
     populate_subscription_purls,
     translate_components,
     trigger_corgi_components_imports
 ]
 
 
-flow_get_corgi_components = [
+flow_get_active_subscriptions = [
     get_config,
-    get_active_subscriptions,
-    populate_components_generator,
-    WHILE(
```

Comment:
I'm worried about removing this - when testing, please make sure the number of components collected are as expected.

The original code attemptps to populate/consume the generator in the `WHILE` statement till it's exhausted. I don't see in this patch how it was dealt with. Assuming there could be missings, i.e., only the first 100 rpm components are processed.

Reply:
Updated.